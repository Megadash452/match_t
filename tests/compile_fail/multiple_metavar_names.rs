fn main() { }

fn my_fn<T: std::any::Any>() {
    match_t::match_t! {
        if T is bool | char {
            println!("{}", core::mem::size_of::<$T>());
            println!("{}", core::mem::size_of::<$G>());
        } else {
            println!("{}", std::any::type_name::<T>());
        }
    }
}
