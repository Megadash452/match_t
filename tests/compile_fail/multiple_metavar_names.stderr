error: Only one Metavariable name can be used within a branch. Found $G while $T exists.
  --> tests/compile_fail/multiple_metavar_names.rs:4:5
   |
4  | /     match_t::match_t! {
5  | |         if T is bool | char {
6  | |             println!("{}", core::mem::size_of::<$T>());
7  | |             println!("{}", core::mem::size_of::<$G>());
...  |
11 | |     }
   | |_____^
   |
   = note: this error originates in the macro `match_t::match_t` (in Nightly builds, run with -Z macro-backtrace for more info)
